streatham_go/app/views/leaderboard.py

```diff
@@ -1,7 +1,42 @@
 from django.shortcuts import render
 from django.contrib.auth.decorators import login_required
+from django.contrib.auth.models import User
+from ..models import Leaderboard
 
 
 @login_required
 def leaderboard(request):
-    return render(request, 'app/leaderboard.html')
+    leaderboard_data = Leaderboard.objects.order_by('-level', 'quiz_count')
+    users = User.objects.all()
+    user_data = []
+    for user in users:
+        user_leaderboard_data = leaderboard_data.filter(user=user).first()
+        user_data.append({
+            'username':
+            user.username,
+            'level':
+            user_leaderboard_data.level if user_leaderboard_data else 1,
+            'quiz_count':
+            user_leaderboard_data.quiz_count if user_leaderboard_data else 0
+        })
+
```

Comment:
Instead of looping through the users and the finding that user in the leaderboard, just loop though the leaderboard.

Also move this into a function in leaderboard class

Reply:
Separate function now, and loops through leaderboard_data as opposed to users